handbook/engineering/distribution/observability/monitoring_architecture.md

```diff
@@ -0,0 +1,114 @@
+
+### Sourcegraph Prometheus
+
+We use [Prometheus](https://prometheus.io) for:
+
+- Collecting high-level, and low-cardinality, metrics from our services.
+- Defining Sourcegraph alerts as both:
+  - Prometheus recording rules, [`alert_count`](#alert-count-metrics).
+  - Prometheus alert rules (which trigger [notifications](#alert-notifications)) based on `alert_count` metrics.
+
+The [`sourcegraph/prometheus`](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus) image handles shipping Sourcegraph metrics and alerting. It bundles:
+
+* Preconfigured [Prometheus](https://prometheus.io), which consumes metrics from Sourcegraph services.
+* Alert and recording rules generated by the [monitoring generator](#monitoring-generator).
+* [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/), which handles alerts from Prometheus.
+* [prom-wrapper](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus/cmd/prom-wrapper), which subscribes to updates in [site configuration](https://docs.sourcegraph.com/admin/config/site_config) and propagates relevant settings to Alertmanager configuration.
+
+#### Alert count metrics
+
+`alert_count` metrics are special Prometheus recording rules that evaluate a single upper or lower bound, as defined in an [Observable](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:generator.go+type+Observable+struct+%7B:%5Bdef%5D%7D&patternType=structural) and generated by the [monitoring generator](#monitoring-generator). This metric is always either 0 if the threshold is not exceeded (or data does not exist, if configured), or 1 if the threshold is exceeded. This allows historical alert data to easily be [consumed programmatically](https://docs.sourcegraph.com/admin/observability/alerting_custom_consumption).
+
+Learn more about the `alert_count` metrics in the [metrics guide](https://docs.sourcegraph.com/admin/observability/metrics_guide#alert-count).
+
+*Rationale for `alert_count`*: TODO(@slimsag)
+
+#### Alert notifications
+
+We use [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) for:
+
+- Providing data about currently active Sourcegraph alerts.
+- Routing alerts to appropriate receivers and silencing them when desired, [configured using site configuration](#alert-notifications).
+
+Alertmanager is bundled in `sourcegraph/prometheus`, and notifications are configured for Sourcegraph alerts [using site configuration](https://docs.sourcegraph.com/admin/observability/alerting). This functionality is provided by the [prom-wrapper](#prom-wrapper).
+
+*Rationale for notifiers in site configuration*: Due to the limitations of [admin reverse-proxies](#admin-reverse-proxy), alerts cannot be configured without port-forwarding or custom ConfigMaps, something we [want to avoid](monitoring_pillars.md#long-term-vision).
```

Comment:
As I recall from our conversation, we could have shipped via `ConfigMaps`, this is a limitation of the implementation we want to do of `siteconfig <- prom-wrapper`. 
I think we should add somewhere that we want to request admins when onboarding a site through the frontend to configure a default destination for notifications, which is driving this implementation, otherwise I think this could also be a required `var` of our deployments.

Reply:
It's twofold - both your point, and [this comment](https://github.com/sourcegraph/about/pull/1221#discussion_r456457481)

(at least, from my understanding 😅 )

Reply:
> we want to request admins when onboarding a site through the frontend to configure a default destination for notifications

https://github.com/sourcegraph/sourcegraph/issues/12332